### PR TITLE
Fix compare_fingerprint errors

### PR DESCRIPTION
**Pull request by albertopasqualetto**
dll fix is not so good, but it works.

### CODE REVIEW

**Comment by sampsyo** [comment]
Hi there! Could you provide a little more context for these changes? What problems were you seeing?

**Comment by albertopasqualetto** [comment]
@sampsyo for the changes in `acoustid.py` `max` function was wrong in python, and for the `[1]` is because a is a tuple and the decode fingerprint only requires the byte array.

For the changes in `chromaprint.py`, i am on windows 11 and `chromaprint.dll` is not an output of the chromaprint build and install (all the functions are in `libchromaprint.dll`), also without the `winmode=0` the dll is not found, i do not know the real reason (i think for some security reason), but I found that piece of code and that works.

File: acoustid.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def _match_fingerprints(a: List[int], b: List[int]) -> float:
             if biterror <= MAX_BIT_ERROR:
                 offset = i - j + bsize
                 counts[offset] += 1
-    topcount = counts.max()
+    topcount = max(counts)
```
Comment (albertopasqualetto) [inline acoustid.py:385]: Correct python's function.

File: chromaprint.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def _guess_lib_name():
     if sys.platform == 'darwin':
         return ('libchromaprint.1.dylib', 'libchromaprint.0.dylib')
     elif sys.platform == 'win32':
-        return ('chromaprint.dll', 'libchromaprint.dll')
+        return ('', 'libchromaprint.dll')
```
Comment (albertopasqualetto) [inline chromaprint.py:28]: Only libchromaprint.dll is created by chromaprint's build, and it contains all the used functions.

File: chromaprint.py [hunk 3]
```diff
@@ -44,7 +44,10 @@ def _load_library(name):
             return None
 
     try:
-        return ctypes.cdll.LoadLibrary(name)
+        if sys.platform == 'win32':
+            return ctypes.WinDLL(name, winmode=0)
+        else:
+            return ctypes.cdll.LoadLibrary(name)
```
Comment (albertopasqualetto) [inline chromaprint.py:50]: It doesn't work without winmode=0, don't know why.

File: acoustid.py [hunk 4]
```diff
@@ -397,8 +397,8 @@ def compare_fingerprints(a, b) -> float:
         raise ModuleNotFoundError("function needs chromaprint")
 
     # decompress fingerprints
-    a = [int(x) for x in chromaprint.decode_fingerprint(a)[0]]
-    b = [int(x) for x in chromaprint.decode_fingerprint(b)[0]]
+    a = [int(x) for x in chromaprint.decode_fingerprint(a[1])[0]]
+    b = [int(x) for x in chromaprint.decode_fingerprint(b[1])[0]]
```
Comment (albertopasqualetto) [inline acoustid.py:401]: `decode_fingerprint` uses the actual fingerprint data, not all the tuple.

**Comment by sampsyo** [comment]
Hello! Since there are several unrelated changes in here, it's going to be important to understand one at a time. In general, while these changes might have made the library work on your setup, we need to proceed super carefully in each case to make sure that we don't break the library for others on different setups—and some of these, such as switching to `WinDLL`, could have implications that we're not anticipating.

One way to move this forward might be to split this PR into several smaller ones so we can discuss them in more detail. The changes to fingerprint comparison, i.e., the follow-up to #75, could be one unit—we can make absolutely sure we're getting the data types right at each point. And then the Windows-related library loading changes could be considered separately.

Thanks for your patience! Just want to make sure we proceed with care.

**Comment by albertopasqualetto** [comment]
Ok, I created 2 different PRs, one for fingerprint comparison (#78 ) and one for Windows import problems (#79).
I am closing this PR.
Thanks for your interest.

**Pull request closed without merging**